api/src/main/java/edu/cornell/mannlib/vitro/webapp/auth/checks/SparqlSelectQueryResultsChecker.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.RDFNode;
+
+public class SparqlSelectQueryResultsChecker {
+    private static final Log log = LogFactory.getLog(SparqlSelectQueryResultsChecker.class);
+
+    public static boolean sparqlSelectQueryResultsContain(Check check, AuthorizationRequest ar, String[] inputValues) {
+        String queryTemplate = check.getConfiguration();
+        if (StringUtils.isBlank(queryTemplate)) {
+            queryTemplate = check.getValues().getSingleValue();
+        }
+        if (StringUtils.isBlank(queryTemplate)) {
+            log.error("SparqlQueryContains template is empty");
+            return false;
+        }
+        AccessObject ao = ar.getAccessObject();
+        Model m = ao.getModel();
+        if (m == null) {
+            log.debug("SparqlQueryContains model is not provided");
+            return false;
+        }
+        Set<String> profileUris = new HashSet<String>(ar.getEditorUris());
+        if (profileUris.isEmpty()) {
+            if (queryTemplate.contains("?profileUri")) {
+                log.debug("Subject has no person URIs");
+                return false;
+            } else {
+                profileUris.add("");
+            }
+        }
+        Set<String> comparedValues = new HashSet<>();
+        if (isQueryNotProvidedInConfiguration(check)) {
+            addRelatedUrisToComparedValues(ao, comparedValues);
+        } else {
+            addValuesToComparedValues(check.getValues(), comparedValues);
+        }
+        for (String profileUri : profileUris) {
+            Set<String> sparqlSelectResults = getSparqlSelectResults(m, profileUri, queryTemplate, ar);
+            // Return true if intersection is not empty
+            comparedValues.retainAll(sparqlSelectResults);
+            if (!comparedValues.isEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static void addValuesToComparedValues(AttributeValueSet values, Set<String> comparedValues) {
+        comparedValues.addAll(values.getValues());
+    }
+
+    private static void addRelatedUrisToComparedValues(AccessObject ao, Set<String> comparedValues) {
+        comparedValues.addAll(Arrays.asList(ao.getResourceUris()));
+    }
+
+    private static boolean isQueryNotProvidedInConfiguration(Check check) {
+        return StringUtils.isBlank(check.getConfiguration());
+    }
+
+    private static Set<String> getSparqlSelectResults(Model model, String profileUri, String queryTemplate,
+            AuthorizationRequest ar) {
+        HashMap<String, Set<String>> queryMap = QueryResultsMapCache.get();
+        String queryMapKey = createQueryMapKey(profileUri, queryTemplate, ar);
+        if (queryMap.containsKey(queryMapKey)) {
+            return queryMap.get(queryMapKey);
+        }
+        Set<String> results = new HashSet<>();
+        ParameterizedSparqlString pss = new ParameterizedSparqlString();
```

Comment:
I wouldn't use an abbreviation here, full name is maybe long but is way more readable.

Reply:
It is a widely used abbreviation. I would leave it this way.

Reply:
ok then, if it is widely used